### PR TITLE
common-instancetypes: Add deployment instructions

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This adds instructions which explain how to deploy common-instancetypes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: docs/operations/deploy_common_instancetypes.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Deploy common-instancetypes
+
+The [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) provide a set of [instancetypes and preferences](../virtual_machines/instancetypes.md) to help create KubeVirt [`VirtualMachines`](http://kubevirt.io/api-reference/main/definitions.html#_v1alpha1_virtualmachine).
```
Comment (cwilkers) [inline docs/operations/deploy_common_instancetypes.md:3]: Why not v1 on the VM? https://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine

File: docs/operations/deploy_common_instancetypes.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Beginning with the 1.1 release of KubeVirt, cluster-wide resources can be deployed directly through KubeVirt, without the need to install another operator.
```
Comment (cwilkers) [inline docs/operations/deploy_common_instancetypes.md:5]: You might need another sentence here to complete the thought, e.g.

"... cluster wide resources can be deployed directly... without another operator..."  This allows deployment of a set of default instance types and preferences along side KubeVirt.

File: docs/operations/deploy_common_instancetypes.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Enabling automatic deployment of common-instancetypes
+
+To enable the deployment of cluster-wide common-instancetypes through KubeVirt respectively `virt-operator`, the `CommonInstancetypesDeploymentGate` feature gate needs to be enabled.
```
Comment (cwilkers) [inline docs/operations/deploy_common_instancetypes.md:9]: Maybe "through the KubeVirt `virt-operator`"

File: docs/operations/deploy_common_instancetypes.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Deploy common-instancetypes
+
+The [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) provide a set of [instancetypes and preferences](../virtual_machines/instancetypes.md) to help create KubeVirt [`VirtualMachines`](http://kubevirt.io/api-reference/main/definitions.html#_v1alpha1_virtualmachine).
+
+Beginning with the 1.1 release of KubeVirt, cluster-wide resources can be deployed directly through KubeVirt, without the need to install another operator.
+
+## Enabling automatic deployment of common-instancetypes
```
Comment (cwilkers) [inline docs/operations/deploy_common_instancetypes.md:7]:
```suggestion
## Enable automatic deployment of common-instancetypes
```

It is a good habit to keep headings in the same tense.

File: docs/operations/deploy_common_instancetypes.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+See [Activating feature gates](activating_feature_gates.md) on how to enable it.
+
+## Deploy common-instancetypes manually
+
+For customization purposes or to install namespaced resources, common-instancetypes can also be deployed by hand.
+
+To install all resources provided by the [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) project without further customizations, simply build with `kustomize` and apply:
+
+```yaml
+$ kubectl kustomize https://github.com/kubevirt/common-instancetypes.git | kubectl apply -f -
```
Comment (cwilkers) [inline docs/operations/deploy_common_instancetypes.md:20]:
```suggestion
$ kubectl apply -k https://github.com/kubevirt/common-instancetypes.git
```

**Review by cwilkers — CHANGES_REQUESTED** [review]
Some small changes, otherwise it looks good

File: docs/operations/deploy_common_instancetypes.md (hunk 1 above)
Comment (0xFelix) [reply docs/operations/deploy_common_instancetypes.md:3]: Good question, that was copied and pasted.

**Comment by cwilkers** [comment]
/hold
For additional reviews, remove when you're all satisfied

**Review by cwilkers — APPROVED** [review]
Looks great!

File: docs/operations/deploy_common_instancetypes.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+To enable the deployment of cluster-wide common-instancetypes through the KubeVirt `virt-operator`, the `CommonInstancetypesDeploymentGate` feature gate needs to be enabled.
+
+See [Activating feature gates](activating_feature_gates.md) on how to enable it.
+
+## Deploy common-instancetypes manually
+
+For customization purposes or to install namespaced resources, common-instancetypes can also be deployed by hand.
+
+To install all resources provided by the [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) project without further customizations, simply apply with `kustomize` enabled:
```
Comment (jcanocan) [inline docs/operations/deploy_common_instancetypes.md:18]: What does "`kustomize` enabled" mean? Could you please clarify?

File: docs/operations/deploy_common_instancetypes.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+```yaml
+$ kubectl apply -k https://github.com/kubevirt/common-instancetypes.git
+```
+
+Alternatively targets for each of the available custom resource types (e.g. namespaced instancetypes) are available.
```
Comment (jcanocan) [inline docs/operations/deploy_common_instancetypes.md:24]:
```suggestion
Alternatively, targets for each of the available custom resource types (e.g. namespaced instancetypes) are available.
```

Comment (jcanocan) [reply docs/operations/deploy_common_instancetypes.md:24]:
```suggestion
Alternatively, there are a variety of targets included for each of the custom resource types available. For instance, namespaced instance types (maybe here we can mention the specific target), name another one, etc.
```

**Review by jcanocan — COMMENTED** [review]
Nice effort @0xFelix ! Just a couple of comments.

**Pull request merged**